[ffmpeg] Add support for libdcadec

popcornmix opened the pull request:
This enables libdcadec support in ffmpeg which means you get the 8 channel, lossless decode of DTS-HD tracks.

Just pushing this out in case anyone would find it useful to cherry-pick for now, but we might want to include it in Isenguard+1

popcornmix committed:
[ffmpeg] Add support for libdcadec

fritsch commented:
@uNiversaI says who?

fritsch commented:
Besides that - I think that all platforms should have the opportunity to use dcadec when we compile it. It's also worth it on linux, windows and so on. 

Let me PR you something.

MartijnKaijser commented:
@fritsch it's DTS-HD as said by the DTS company

fritsch commented:
@MartijnKaijser I know - but there are other things to be discussed here first. I will open the BikeShedding Process with a green lamp, when ready.

fritsch commented:
via: https://github.com/popcornmix/xbmc/pull/11

uNiversaI commented:
@fritsch 

You have currently this sentence

```
+msgid "This option supports 8 channel DTS HD decoding, but may use more CPU. It is only available when DTS and DTS-HD audio passthrough is disabled"
```

You use both **DTS HD*** and ***DTS-HD*** on same sentence.
Also All existing entries refer only to ***DTS-HD*** only https://github.com/xbmc/xbmc/search?utf8=%E2%9C%93&q=DTS-hd

So why are we causing more inconsistencies?

da-anda commented on addons/resource.language.en_gb/resources/strings.po:
```diff
@@ -16541,3 +16541,13 @@ msgstr ""
+
+#: system/settings/rbp.xml system/settings/imx6.xml
+msgctxt "#38120"
+msgid "Support 8 channel DTS HD audio decoding"
```
I'm not a friend of long settings labels with probably more technical info than needed, because they tend to get even longer in translations and then scroll. Would "Enable DTS-HD audio decoding" also work? If it's 8 channel or not is of no value for mortals I think and that additional technical info is well suited in the description.

uNiversaI replied:
Agree with da-anda's assessment.

fritsch replied:
@da-anda: PR welcome.
Am 10.05.2015 11:59 vorm. schrieb "uNiversaI" <notifications@github.com>:

> In addons/resource.language.en_gb/resources/strings.po
> <https://github.com/xbmc/xbmc/pull/7102#discussion_r30000554>:
>
> > @@ -16541,3 +16541,13 @@ msgstr ""
> >  msgctxt "#38016"
> >  msgid "%d fps"
> >  msgstr ""
> > +
> > +#: system/settings/rbp.xml system/settings/imx6.xml
> > +msgctxt "#38120"
> > +msgid "Support 8 channel DTS HD audio decoding"
>
> Agree with da-anda's assessment.
>
> --
> Reply to this email directly or view it on GitHub
> <https://github.com/xbmc/xbmc/pull/7102/files#r30000554>.
>

da-anda replied:
I'm not THE entity for labels, so I only give suggestions. If you think they are fine I'll PR

fritsch replied:
Yeah - but you know what is best - I don't. I am always thankful for
comments, but sometimes - I am in a hurry and don't have the time. We
reworked some other language settings in the past together, which was quite
succesful, you remember?

If you give me two strings, I will PR them.

2015-05-10 14:31 GMT+02:00 da-anda <notifications@github.com>:

> In addons/resource.language.en_gb/resources/strings.po
> <https://github.com/xbmc/xbmc/pull/7102#discussion_r30001665>:
>
> > @@ -16541,3 +16541,13 @@ msgstr ""
> >  msgctxt "#38016"
> >  msgid "%d fps"
> >  msgstr ""
> > +
> > +#: system/settings/rbp.xml system/settings/imx6.xml
> > +msgctxt "#38120"
> > +msgid "Support 8 channel DTS HD audio decoding"
>
> I'm not THE entity for labels, so I only give suggestions. If you think
> they are fine I'll PR
>
> --
> Reply to this email directly or view it on GitHub
> <https://github.com/xbmc/xbmc/pull/7102/files#r30001665>.
>



-- 
                   Key-ID:     0x1A995A9B
                   keyserver: pgp.mit.edu
==============================================================
Fingerprint: 4606 DA19 EC2E 9A0B 0157  C81B DA07 CF63 1A99 5A9B

wsnipex commented on configure.ac:
```diff
@@ -1088,6 +1088,7 @@ AC_CHECK_HEADER([mpeg2dec/mpeg2convert.h],, AC_MSG_ERROR($missing_library),
 AC_CHECK_HEADER([jpeglib.h],,        AC_MSG_ERROR($missing_library))
 AC_CHECK_HEADER([ogg/ogg.h],,        AC_MSG_ERROR($missing_library))
 AC_CHECK_HEADER([vorbis/vorbisfile.h],, AC_MSG_ERROR($missing_library))
+AC_CHECK_HEADER([libdcadec/dca_context.h],, AC_MSG_ERROR($missing_library))
```
can we use PKG_CHECK_MODULES here? And please replace $missing_library with a more descriptive message, e.g. "libdcadec not found"

wsnipex commented on tools/depends/target/libdcadec/Makefile:
```diff
@@ -0,0 +1,38 @@
+# lib name, version
+LIBNAME=dcadec
+ARCHIVE=master.tar.gz
+BASE_URL=https://github.com/foo86/dcadec/archive
```
please upload a tagged version to our mirrors instead. Let me know if I should do that for you.

wsnipex commented on tools/depends/target/libdcadec/Makefile:
```diff
@@ -0,0 +1,38 @@
+	cd $(PLATFORM); $(ARCHIVE_TOOL) $(ARCHIVE_TOOL_FLAGS) $(TARBALLS_LOCATION)/$(ARCHIVE)
+
+$(LIBDYLIB): $(PLATFORM)
+	$(MAKE) -C $(PLATFORM) PREFIX=$(PREFIX) CC="$(CC)" CFLAGS="$(CFLAGS)" LDFLAGS="$(LDFLAGS)" AR="$(AR)"
```
are all those compiler flags needed? They should all be set already by Makefile.include

wsnipex commented:
Please don't add ffmpeg patches in depends, they should go to our ffmpeg repo instead(or come in automatically with the next rebase)

da-anda commented on addons/resource.language.en_gb/resources/strings.po:
```diff
@@ -16541,3 +16541,13 @@ msgstr ""
+
+#: system/settings/rbp.xml system/settings/imx6.xml
+msgctxt "#38121"
+msgid "This option supports 8 channel DTS HD decoding, but may use more CPU. It is only available when DTS and DTS-HD audio passthrough is disabled"
```
We changed all audio related settings to be a bit more user friendly and also point out the benefit a user will have in a less technical way. So maybe we can rephrase this a bit as well, like pointing out that you're able to listen to DTS-HD even if your AVR doesn't support it? This would be my take on it, but feel free to change it:

"Enables decoding for DTS-HD audio streams. This allows conversion of DTS-HD audio into a format supported by the connected audio device and will give lossless 8 channel audio fidelity on devices that support PCM. Note that this might increase CPU load and is only available when DTS and DTS-HD audio passthrough is disabled."

PS: these labels are also used in pi.xml and pi2.xml files for which the useage comment is missing. A small contextual note would also be nice on both labels, like "Name of a setting" and "Description of setting #38120 "Foo-> Bar -> Enable DTS-HD audio decoding"

da-anda replied:
@fritsch does this description sound good to you from a technical point or are there errors?
```
Enables conversion of DTS-HD audio streams into a format supported by the connected audio device and will give bitperfect 8 channel audio fidelity on devices that support PCM. Note that this might increase CPU load and is only available when DTS and DTS-HD audio passthrough are disabled.
```
What I wonder though is, why does DTS-HD decoding depend on DTS passthrough? Because DTS-HD does also contain a DTS core stream? What about AVRs that support DTS but not DTS-HD - can't I have DTS passthrough for videos with DTS track but decoding to PCM when I have a DTS-HD track?

fritsch replied:
It's not correct.

What about:
"Enables decoding of high quality DTS-HD audio streams. Note that this
increases CPU load and is only available when DTS and DTS-HD audio
passthrough are disabled."

@da-anda: Yes. It's currently one vs the other.

2015-05-10 17:35 GMT+02:00 da-anda <notifications@github.com>:

> In addons/resource.language.en_gb/resources/strings.po
> <https://github.com/xbmc/xbmc/pull/7102#discussion_r30003149>:
>
> > @@ -16541,3 +16541,13 @@ msgstr ""
> >  msgctxt "#38016"
> >  msgid "%d fps"
> >  msgstr ""
> > +
> > +#: system/settings/rbp.xml system/settings/imx6.xml
> > +msgctxt "#38120"
> > +msgid "Support 8 channel DTS HD audio decoding"
> > +msgstr ""
> > +
> > +#: system/settings/rbp.xml system/settings/imx6.xml
> > +msgctxt "#38121"
> > +msgid "This option supports 8 channel DTS HD decoding, but may use more CPU. It is only available when DTS and DTS-HD audio passthrough is disabled"
>
> @fritsch <https://github.com/fritsch> does this description sound good to
> you from a technical point or are there errors?
>
> Enables conversion of DTS-HD audio streams into a format supported by the connected audio device and will give bitperfect 8 channel audio fidelity on devices that support PCM. Note that this might increase CPU load and is only available when DTS and DTS-HD audio passthrough are disabled.
>
> What I wonder though is, why does DTS-HD decoding depend on DTS
> passthrough? Because DTS-HD does also contain a DTS core stream? What about
> AVRs that support DTS but not DTS-HD - can't I have DTS passthrough for
> videos with DTS track but decoding to PCM when I have a DTS-HD track?
>
> --
> Reply to this email directly or view it on GitHub
> <https://github.com/xbmc/xbmc/pull/7102/files#r30003149>.
>



-- 
                   Key-ID:     0x1A995A9B
                   keyserver: pgp.mit.edu
==============================================================
Fingerprint: 4606 DA19 EC2E 9A0B 0157  C81B DA07 CF63 1A99 5A9B

uNiversaI replied:
Nice, here is a small suggestion to match the [strings] work Im doing atm.
```
"Enables decoding of high quality DTS-HD audio streams. Note: This
increases CPU load and is only available when DTS and DTS-HD audio
passthrough are disabled."
```

jjd-uk commented:
I'm curious why a new setting is being introduced? as we don't have a similar setting for TrueHD, we simply automatically decode to PCM if TrueHD is not enabled. I'm asking as in my view the methods for TrueHD and DTS-HD should be consistent from a UI point of view.

fritsch commented:
Reason: IMX (2 core), Raspberry Pi A, B+ might be too slow to get that decoded.

On all other platforms it is default enabled.

popcornmix commented:
Squashed existing commits.

@wsnipex Looks like there are no tags on https://github.com/foo86/dcadec
and it's pretty active. Milhouse builds have been rebuilding with latest package every day for about six weeks with no reports of regressions so it seems quite stable. If you can upload a version (e.g. from now) as a point in the sand, I can make the PR use that, so we don't forget.

Agreed the first commit should be dropped and added to our ffmpeg repo. Depending on when this gets merged one of our ffmpeg bumps might pick it up (the commits are from master of ffmpeg).

fritsch commented:
@popcornmix we need something additional here? What is needed to get that one in?

popcornmix commented:
I think it's virtually there. The ffmpeg patches have been dropped (they are in current master version).
We are currently using this as the source: 
https://github.com/foo86/dcadec/archive/master.tar.gz

I suspect we need to pick a version and host it ourselves.
There are no releases/tags for libdcadec, but as it hasn't changed for six weeks and as the current version seems fine (e.g. in Milhouse build) I'd suggest we just grab the current master version.

If someone can upload the tarball to a suitable place I can update the URL in the makefile.

wsnipex commented:
what happend to the changes from https://github.com/popcornmix/xbmc/pull/14

popcornmix commented:
@wsnipex they were there... Let me have a look.

popcornmix commented:
Okay, I've added back the @wsnipex commits.
Might be worth bumping the version - there are a couple of dozen commits since 396e75652.

fritsch commented:
You mean bumping ffmpeg? and then we are done?

popcornmix commented:
No, ffmpeg is okay. The tarball from https://github.com/foo86/dcadec is a little old.

MartijnKaijser commented on tools/depends/target/ffmpeg/Makefile:
```diff
@@ -56,6 +56,7 @@ endif
 ifeq ($(Configuration), Release)
   ffmpg_config += --disable-debug
 endif
+ffmpg_config += --enable-libdcadec
```
why not put this on line 22 instead of behind all the ifdef? Any build issues?

fritsch commented:
@popcornmix thx. @wsnipex could you bump the tar? then we could continue this here to get it merged?

wsnipex commented:
current master tarball uploaded to mirrors as libdcadec-git-2a9186e3.tar.gz
might take a bit to propagate.

popcornmix commented:
Updated to use 2a9186e3.

fritsch commented:
Oki - will start a testbuild tomorrow morning, if no one is faster!

Thanks much to you both.

popcornmix commented:
Moved the ffmpg_config line up as suggested by @MartijnKaijser

wsnipex commented:
jenkins build this please

MartijnKaijser commented:
jenkins build this please

koying commented:
/cc me

fritsch commented:
jenkins build this please

stefansaraev commented:
jenkins build this please again

mkortstiege commented:
jenkins build this please

fritsch commented:
That needs some love on OSX-32. Seems the linker cannot find libdcadec objects. I sadly cannot see the reason for the linux fail anymore (Groovy script failed).

Memphiz commented on tools/depends/target/Makefile:
```diff
@@ -96,7 +96,7 @@ nettle: gmp
 libsdl2: $(LINUX_SYSTEM_LIBS)
 libxslt: libgcrypt
-ffmpeg: $(ICONV) $(ZLIB) bzip2 libvorbis $(FFMPEG_DEPENDS)
+ffmpeg: $(ICONV) $(ZLIB) bzip2 libvorbis libdcadec $(FFMPEG_DEPENDS)
```
Might be ok to add it to the FFMPEG_DEPENDS variable instead? (cause its only used by it if i see it right)

Memphiz commented on tools/depends/target/libdcadec/Makefile:
```diff
@@ -0,0 +1,39 @@
+
+# configuration settings
+
+LIBDYLIB=$(PLATFORM)/.libs/lib$(LIBNAME).a
```
this expands to liblibdcadec.a ... drop the extra lib here

Memphiz commented on tools/depends/target/libdcadec/Makefile:
```diff
@@ -0,0 +1,39 @@
+
+LIBDYLIB=$(PLATFORM)/.libs/lib$(LIBNAME).a
+
+CLEAN_FILES=$(ARCHIVE) $(PLATFORM)
```
unused

Memphiz commented on tools/depends/target/libdcadec/Makefile:
```diff
@@ -0,0 +1,39 @@
+
+CLEAN_FILES=$(ARCHIVE) $(PLATFORM)
+
+CFLAGS += -std=gnu99
```
not sure - but do we need this? From the upper layer of the depends buildsystem it normally uses c++11 standard (i don't know how the compiler behaves if both flags are set) - this might even be a problem on osx because the syntax is different ...

Memphiz commented on xbmc/cores/dvdplayer/DVDCodecs/Audio/DVDAudioCodecFFmpeg.cpp:
```diff
@@ -54,10 +55,19 @@ CDVDAudioCodecFFmpeg::~CDVDAudioCodecFFmpeg()
 
-  pCodec = avcodec_find_decoder(hints.codec);
+  bool allow_dtshd_decoding = true;
+#if defined(TARGET_RASPBERRY_PI) || defined(HAS_IMXVPU)
```
Why is it restricted to those platforms? We don't need to build dcadec for all platforms if it is only used for 2 ...

Memphiz replied:
ahh i see - only on those it is a setting - right? its default on on all others?

Memphiz commented:
mhhh the linker error  on osx and ios is because it needs to be added here 

https://github.com/xbmc/xbmc/blob/master/tools/darwin/Configurations/App.xcconfig.in#L28

seems there is no problem with building the libdcadec (so discard my std99 comment)

koying commented on docs/README.ubuntu:
```diff
@@ -82,7 +82,7 @@ Tip: For those with multiple computers at home is to try out distcc
 For Ubuntu (all versions >= 7.04):
 
-    $ sudo apt-get install automake bison build-essential cmake curl cvs default-jre fp-compiler gawk gdc gettext git-core gperf libasound2-dev libass-dev libboost-dev libboost-thread-dev libbz2-dev libcap-dev libcdio-dev libcurl3 libcurl4-gnutls-dev libdbus-1-dev libfontconfig-dev libfreetype6-dev libfribidi-dev libgif-dev libglew-dev libiso9660-dev libjasper-dev libjpeg-dev liblzo2-dev libmicro ...
+    $ sudo apt-get install automake bison build-essential cmake curl cvs dcadec-dev default-jre fp-compiler gawk gdc gettext git-core gperf libasound2-dev libass-dev libboost-dev libboost-thread-dev libbz2-dev libcap-dev libcdio-dev libcurl3 libcurl4-gnutls-dev libdbus-1-dev libfontconfig-dev libfreetype6-dev libfribidi-dev libgif-dev libglew-dev libiso9660-dev libjasper-dev libjpeg-dev liblzo2-d ...
```
@wsnipex, looks like the package name in ubuntu is libdca-dev, isn't it?

wsnipex replied:
no, that one is http://www.videolan.org/developers/libdca.html without HD capabilities

koying replied:
ok. I don't find dcadec-dev in trusty, though, nor any mention of this package on internet beside launchpad builds.
Might just be my eyes ;)

wsnipex replied:
corect, there is none, thats why I packaged it for the ppa

koying commented:
Is it just me or if DTS passthrough and 7.1 speakers are enabled, DTS master passthrough (5.1) is used rather than 7.1 PCM? 
If I disable passthrough, works fine..
(on droid/shield)

fritsch commented:
Yes - that's the intention of this PR.

2015-07-31 19:48 GMT+02:00 Chris Browet <notifications@github.com>:

> Is it just me or if DTS passthrough and 7.1 speakers are enabled, DTS
> master passthrough (5.1) is used rather than 7.1 PCM?
> If I disable passthrough, works fine..
> (on droid/shield)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/xbmc/xbmc/pull/7102#issuecomment-126769717>.
>



-- 
                   Key-ID:     0x1A995A9B
                   keyserver: pgp.mit.edu
==============================================================
Fingerprint: 4606 DA19 EC2E 9A0B 0157  C81B DA07 CF63 1A99 5A9B

popcornmix commented:
Yes, help test says:
"Enables decoding of high quality DTS-HD audio streams. Note: This increases CPU load and is only available when DTS and DTS-HD audio passthrough are disabled."

koying commented:
Seems counter-intuitive to me, especially if 7.1 is explicitly selected.
Expect a sh...oad of support tickets ;)

fritsch commented:
Keep them coming :-) Besides that, does it work as expected?

koying commented:
Yep, great :)

MartijnKaijser commented:
How on earth are non RPi/imx6 users gonna see the help text which tells them what is happening if it's not visible on their platform?

koying commented:
Haha... Good point :)

popcornmix commented:
Yep. There's always the wiki.

MartijnKaijser commented:
yeah right.....

koying commented:
Yeah. Zero chance they'll look into the wiki.
I'd rather have the setting enabled for everybody (or at least droid), then. That makes it explicit, and anyway, if indeed decoding is cpu-intensive, low-specs droid devices might suffer, too.

MartijnKaijser commented:
or low spec whatever devices?

fritsch commented:
shed, shed, shed, shed, shed ... will PR something so it lands in "global" settings.xml

fritsch commented:
https://github.com/fritsch/xbmc/commit/4e7d1334644b3c9ab56f92dd86a3ddb0f08423f5

^^ @popcornmix need to go for lunch - but that should do the job

koying commented:
@fritsch Your patch does what it should but if a user has previously enabled passthrough, as the dts-ma is disabled, he cannot see the help message explaining why he cannot enable it ;)

I'd suggest to add to the passthrough help message that enabling it prevents enabling dts-ma

fritsch commented:
@koying I'd keep it disabled, especially as Vero and Pi1 and other slow droids cannot use it by default. So sane defaults in the sense of: Don't enable something, that might not work.

fritsch commented:
jenkins build this please

popcornmix commented:
Rebased. jenkins build this please

fritsch commented:
@memphiz could you have a look. It seems ffmpeg on IOS is not build with dcadec support?

Memphiz commented:
I repeat my comment from 7 days ago:

mhhh the linker error on osx and ios is because it needs to be added here 

https://github.com/xbmc/xbmc/blob/master/tools/darwin/Configurations/App.xcconfig.in#L28

seems there is no problem with building the libdcadec (so discard my std99 comment) - its built - but not linked in ...

popcornmix commented:
@Memphiz so add -ldcadec to end of that line?

Memphiz commented:
@popcornmix ack

Memphiz commented:
@popcornmix keep it in the near of the other ffmpeg libs in there if you don't mind :)

popcornmix commented:
Updated.

stefansaraev commented:
jenkins build this please again

I think android is going to have some issues :)

MartijnKaijser commented:
Anything missing before this can be merged (besides the squash me commits)?

popcornmix commented:
I've squashed it down to a single commit (without any changes). I couldn't see a good reason for having it split further.

popcornmix commented:
Squashed the assignment to true, then overwriting of allow_dtshd_decoding into one line as suggested by @fritsch

stefansaraev commented:
@popcornmix settings constants after #7721 ?

MartijnKaijser commented on xbmc/cores/dvdplayer/DVDCodecs/Audio/DVDAudioCodecFFmpeg.cpp:
```diff
@@ -33,6 +33,7 @@ extern "C" {
 #include "settings/Settings.h"
 #include "cores/AudioEngine/Utils/AEUtil.h"
 #endif
+#include "settings/Settings.h"
```
You might want to remove the settings.h from the ifdef above

popcornmix commented:
Switched to use settings constants and removed the extra settings.h include.

The Android build issue I don't know much about. So Android doesn't like _BSD_SOURCE being set?
Can anyone suggest a way of removing that define on Android only?

stefansaraev commented:
maybe something like https://github.com/stefansaraev/xbmc/commit/386c6d4e984073fa8c6d96dda4fe4cfcfe2fb90c but let's build test first, just in case. jenkins build this please

hudokkow commented:
jenkins build this please

popcornmix commented:
Android arm passes, but Android x86 fails. I've added @stefansaraev's hack just to see if that is sufficient. jenkins build this please

popcornmix commented:
Rebased. jenkins build this please

koying commented on tools/depends/target/libdcadec/libdcadec_android.patch:
```diff
@@ -0,0 +1,14 @@
+         && sync != SYNC_WORD_CORE_LE14
+         && sync != SYNC_WORD_CORE_BE14) {
+-        int c = DCA_FGETC(stream->fp);
++        int c = fgetc(stream->fp);
```
Might be cleaner to alter the definition of DCA_FGETC rather than replacing it.
If DCA_FGETC  gets reused in further versions, we'd fail again...

stefansaraev replied:
yep indeed. this was quick and dirty blind try

stefansaraev commented:
jenkis hates me. jenkins build this please. again.

stefansaraev commented:
argh. I am blind. sorry. https://github.com/stefansaraev/xbmc/commit/1c6c03ba14a0db81a338a70d024887498dda94ff this should be better. you may squash down all to one commit @popcornmix

popcornmix commented:
squashed in @stefansaraev's latest patch. jenkins build this please

Paxxi commented:
Spoke with @stefansaraev in IRC and agreed to do the win32 work, tonight if time permits or over the weekend

Paxxi committed:
Added prebuilt libdcadec from same source as linux version

Paxxi commented:
@popcornmix pick Paxxi@4fad9d8c61c43ae5419fa0306b17f4c2b41e8e1a for win32 work.
someone needs to upload http://delusional.nu/libdcadec-git-396e75652-win32.zip to build-deps/win32/mingw-msys/ and it should be good to go. Not runtime tested as I don't have anything to test with but ffmpeg built fine

popcornmix commented:
@Paxxi added your commit to PR.

MartijnKaijser commented:
file uploaded. will take some time to be available

stefansaraev commented:
jenkins build this please

stefansaraev commented:
mind rebasing ?

MartijnKaijser commented:
jenkins build this please

popcornmix commented:
I updated the string numbers and added a comment about where the settings strings were from during the rebase, so would be useful if someone just sanity checks that looks okay.

stefansaraev commented:
strings look good to me @popcornmix thanks.

popcornmix commented:
Android-x86 and win32 failed, but doesn't look related to this PR.

stefansaraev commented:
jenkins error unrelated. triggered manual builds (just in case) - all gree. @MartijnKaijser your button

metaron-uk commented:
I've a problem:
```
andrew@blackbox ~/saturn/ksoo-kodi/kodi $ git checkout master && git pull
andrew@blackbox ~/saturn/ksoo-kodi/kodi $ git clean -xfd
andrew@blackbox ~/saturn/ksoo-kodi/kodi $ ./bootstrap && ./configure
...
checking ogg/ogg.h presence... yes
checking for ogg/ogg.h... yes
checking vorbis/vorbisfile.h usability... yes
checking vorbis/vorbisfile.h presence... yes
checking for vorbis/vorbisfile.h... yes
checking for LIBDCADEC... no
configure: error: libdcadec not found
andrew@blackbox ~/saturn/ksoo-kodi/kodi $ make -C tools/depends/target/libdcadec PREFIX=/usr/local
make: Entering directory '/mnt/saturn/home/andrew/ksoo-kodi/kodi/tools/depends/target/libdcadec'
Makefile:1: ../../Makefile.include: No such file or directory
make: *** No rule to make target '../../Makefile.include'.  Stop.
make: Leaving directory '/mnt/saturn/home/andrew/ksoo-kodi/kodi/tools/depends/target/libdcadec'
```
any clues?

MilhouseVH commented:
> any clues?

On Ubuntu I added the nightly PPA:

https://launchpad.net/~team-xbmc/+archive/ubuntu/xbmc-nightly

then `apt-get install dcadec-dev`.

metaron-uk commented:
My solution: 
```
cd tools/depends
su
./configure --with-toolchain=/usr --prefix=/usr/local
cd ../..
make -C tools/depends/target/libdcadec PREFIX=/usr/local
```
(Gentoo's media-sound/dcadec package isn't ready for primetime yet: https://bugs.gentoo.org/buglist.cgi?quicksearch=dcadec )

uNiversaI commented on addons/resource.language.en_gb/resources/strings.po:
```diff
@@ -17580,4 +17580,15 @@ msgstr ""
+#. Description of setting #38118 Settings -> System -> Audio output -> Support 8 channel DTS-HD audio decoding
+#: system/settings/settings.xml
+msgctxt "#38119"
+msgid "Enables decoding of high quality DTS-HD audio streams. Note: This increases CPU load and is only available when DTS and DTS-HD audio passthrough are disabled."
```
You never see this message until  DTS and DTS-HD audio passthrough are disabled.

So perhaps the DTS and DTS-HD audio passthrough should actually include a mention that in order for this option to work you must disable them else the above message is pointless.

fritsch replied:
Not really. Remember passthrough is off by default.
Am 05.09.2015 2:53 nachm. schrieb "uNiversaI" <notifications@github.com>:

> In addons/resource.language.en_gb/resources/strings.po
> <https://github.com/xbmc/xbmc/pull/7102#discussion_r38810503>:
>
> > @@ -17580,4 +17580,15 @@ msgstr ""
> >  #: system/settings/settings.xml
> >  msgctxt "#38017"
> >  msgid "Region A - Americas, East Asia and Southeast Asia. Region B - Africa, Middle East, Southwest Asia, Europe, Australia, New Zealand. Region C - Central Asia, mainland China, Mongolia, South Asia, Belarus, Russia, Ukraine, Kazakhstan."
> > +
> > +#. Setting #38118 "Settings -> System -> Audio output -> Support 8 channel DTS-HD audio decoding"
> > +#: system/settings/settings.xml
> > +msgctxt "#38118"
> > +msgid "Support 8 channel DTS-HD audio decoding"
> > +msgstr ""
> > +
> > +#. Description of setting #38118 Settings -> System -> Audio output -> Support 8 channel DTS-HD audio decoding
> > +#: system/settings/settings.xml
> > +msgctxt "#38119"
> > +msgid "Enables decoding of high quality DTS-HD audio streams. Note: This increases CPU load and is only available when DTS and DTS-HD audio passthrough are disabled."
>
> You never see this message until DTS and DTS-HD audio passthrough are
> disabled.
>
> So perhaps the DTS and DTS-HD audio passthrough should actually include a
> mention that in order for this option to work you must disable them else
> the above message is pointless.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/xbmc/xbmc/pull/7102/files#r38810503>.
>

uNiversaI replied:
Not for current users who have passthrough enabled.

FernetMenta replied:
I did not agree on having this silly setting in global. This is just another step towards even more mess with settings. There is absolutely no reason for the majority of devices to have this disabled. For the other either it works or not so this can be set as an appliance for this low end devices.
In addition pass though is an option to PCM audio. Following this logic giving pass through priority is wrong.

FernetMenta commented on xbmc/cores/dvdplayer/DVDDemuxers/DVDDemuxFFmpeg.cpp:
```diff
@@ -439,6 +439,19 @@ bool CDVDDemuxFFmpeg::Open(CDVDInputStream* pInput, bool streaminfo, bool filein
 
   if (m_streaminfo)
   {
+    if (CSettings::GetInstance().GetBool(CSettings::SETTING_AUDIOOUTPUT_SUPPORTSDTSHDCPUDECODING))
```
this makes no sense

fritsch replied:
It only makes sense if user decides to use dcadec ... if not the streaminfo is not what he is actually getting as plain dts is decoded and therefore 6 channels have to be displayed ...

FernetMenta replied:
you are wrong. this is also used to extract metadata displayed in the library.

fritsch replied:
> EDIT: and also users with passthrough want to see their audio tracks labelled DTS HD MasterAudio

This is your opinion, yes. Mine is / was different.

And concerning what I said: No - I am not wrong, what I said was and is still perfectly right, if we want it that way is a complete different topic - guess what - I don't care at all, it can be done like the one ontop of the roof or like it is under the table of the hemple family ... 

Btw. the setting was introduced only, cause others asked for it (low performance platform whatever), see the history of this PR. It is only there, cause I wanted to speed up integration and went for the compromise ...

FernetMenta replied:
that's the inevitable consequences if such decisions are taken by non-maintainers and even non-devs. I see the history, you changed it after a comment by @MartijnKaijser who has closed PR https://github.com/xbmc/xbmc/pull/4586 in favor of a fix in ffmpeg. Now we have the change in ffmpeg 4586 but still not working because of the reasons I have given.

fritsch replied:
Don't mix two things here.

We still have platforms that are too slow to do full dts-hd decoding - for example the 2 core vero ... those systems _need_ a setting to disable this decoding. Also the raspberry Pi1 needs this setting.

Then there are some single core ION-1 celerons among us, that need it. The settings makes sense (!).

It makes - as you said - no sense, that we _now_ with dcadec support don't show user the real content - this I will patch out in a minute.

But, hey, as we are on argumenting: Can you name me one platform where such a setting would make sense? I know you can - and therefore this setting makes sense ...

FernetMenta replied:
You seem to know already that vero, Pi1 or ION-1 are too slow. Why this setting? IMO much better to disable it on those platforms automatically.

fritsch replied:
Yes, that would be better in deed. What about the users that want to save "decoding" power and therefore only want to decode 5.1 dts instead of the much more demanding dts-hd? Especially when output is 2 channel only?

Think of the TV 2.0 watching users only ... no need at all to decode 7.1 dts-hd and then downmix to 2.0 channels ... default is disabled btw. ... to care for those user automatically.

https://github.com/fritsch/xbmc/commit/e2a3350a6108d344ffe49819d7de6d0ec1163d97?w=0

fritsch replied:
Summary:

Reasons for this setting:
- Users have 2.0 output only and don't care at all if dts-hd 7.1 is decoded and downmixed afterwards
- With the setting we can keep it disabled by default, non hifi users won't realize anything
- Low performance devices (on android, windows (old dual core machines)) can enable it on a per file basis (*)

* Not automatically detectable without high error margin

metaron-uk commented:
Unfortunately I still can't get configure to find libdecadec (see http://forum.kodi.tv/showthread.php?tid=229552&pid=2098454#pid2098454)

fritsch commented:
Sorry - this is not place to answer gentoo build system issues. The Ubuntu way, that we support is supported via our ppa.

wsnipex commented on configure.ac:
```diff
@@ -1139,6 +1140,7 @@ PKG_CHECK_MODULES([GNUTLS], [gnutls], [have_gnutls=yes];AC_DEFINE([HAVE_GNUTLS],
 AC_CHECK_LIB([bz2],         [main],, AC_MSG_ERROR($missing_library))
 AC_CHECK_LIB([jpeg],        [main],, AC_MSG_ERROR($missing_library)) # check for cximage
 AC_CHECK_LIB([tiff],        [main],, AC_MSG_ERROR($missing_library))
+AC_CHECK_LIB([dcadec],      [main],, AC_MSG_ERROR($missing_library))
```
this looks like a bad rebase, this line has to be removed.

wsnipex commented on tools/depends/target/libdcadec/Makefile:
```diff
@@ -0,0 +1,40 @@
+include ../../Makefile.include
```
I remember to have fixed this one as well. should be:
>-include ../../Makefile.include
note the literal "-"
